### PR TITLE
Update shard_map.md's API specification section

### PR DESCRIPTION
**Pull request by yashk2810**

### CODE REVIEW

File: docs/notebooks/shard_map.ipynb [hunk 1]
```diff
@@ -827,17 +827,20 @@
...
     ") -> Callable:\n",
     "  ...\n",
     "```\n",
     "where:\n",
     "* communication collectives like `psum` in the body of `f` can mention the axis names of `mesh`;\n",
-    "* `mesh` encodes devices arranged in an array and with associated axis names, just like it does for `sharding.NamedSharding`;\n",
-    "* `in_specs` and `out_specs` are `PartitionSpec`s which can affinely mention axis names from `mesh` to express slicing/unconcatenation and concatenation of inputs and outputs, respectively, with unmentioned names corresponding to replication and untiling (assert-replicated-so-give-me-one-copy), respectively;\n",
-    "* `auto` is an optional set of axis names corresponding to the subset of names of `mesh` to treat automatically in the body, as in the caller, rather than manually;\n",
+    "* `mesh` encodes devices arranged in an array and with associated axis names, just like it does for `sharding.NamedSharding`; If None, mesh will be inferred from the\n",
+    "context which can be set via `jax.sharding.use_mesh` context manager.\n",
```
Comment (mattjj) [inline docs/notebooks/shard_map.ipynb:840]: "via" -> "via the"

File: docs/notebooks/shard_map.ipynb [hunk 2]
```diff
@@ -827,17 +827,20 @@
...
+    "* `in_specs` are `PartitionSpec`s which can affinely mention axis names from `mesh` to express slicing/unconcatenation of inputs, respectively, with unmentioned names corresponding to replication and untiling (assert-replicated-so-give-me-one-copy). If None, all mesh axes must be of type `Explicit`, in which case the in_specs are inferred from the argument types;\n",
```
Comment (mattjj) [inline docs/notebooks/shard_map.ipynb:841]: "affinely" is such a pretentious word to use here lol, sorry

you don't have to fix it here but we should say "zero or one times" instead

**Review by mattjj — APPROVED** [review]
Beautiful, thanks. And you made it a PR just for me <3

File: docs/notebooks/shard_map.ipynb (hunk 1 above)
Comment (yashk2810) [reply docs/notebooks/shard_map.ipynb:840]: Done

File: docs/notebooks/shard_map.ipynb (hunk 2 above)
Comment (yashk2810) [reply docs/notebooks/shard_map.ipynb:841]: Done
